stabilizer/src/main/java/com/hazelcast/stabilizer/worker/base/BaseWorkerTask.java

```diff
@@ -0,0 +1,77 @@
+package com.hazelcast.stabilizer.worker.base;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.stabilizer.test.TestContext;
+import com.hazelcast.stabilizer.test.annotations.Performance;
+import com.hazelcast.stabilizer.worker.OperationSelector;
+
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Base worker class which is returned by {@link com.hazelcast.stabilizer.test.annotations.RunWithBaseWorker} annotated test
+ * methods.
+ * <p/>
+ * Implicitly logs and measures performance. The related properties can be overwritten with the properties of the test.
+ * The Operation counter is automatically increased after each doRun() call.
+ *
+ * @param <O> Type of Enum used by the {@link com.hazelcast.stabilizer.worker.OperationSelector}
+ */
+public abstract class BaseWorkerTask<O extends Enum<O>> implements Runnable {
```

Comment:
I would call this AbstractWorkerTask since it is abstract, and not prefix it with Base. The user can call his implementation WorkerTask which is nice and clean.